### PR TITLE
Data-provider-desc-sort-prio

### PR DESCRIPTION
**Pull request by jdroenner**
- [ ] I added an entry to [`CHANGELOG.md`](CHANGELOG.md) if knowledge of this change could be valuable to users.

---

Here is a brief summary of what I did:

<TEXT>

### CODE REVIEW

**Comment by jdroenner** [comment]
[UI Clean-Up + Dataset Tag Listing](https://trello.com/c/OafKazTf/1529-ui-clean-up-dataset-tag-listing)

File: services/src/contexts/migrations/migration_0003_gbif_config.rs [hunk 1]
```diff
@@ -106,11 +102,13 @@ mod tests {
         // drop the connection because the pool is limited to one connection, s.t. we can reuse the temporary schema
         drop(conn);
 
+        /* Note: this only works if the rust type is
```
Comment (michaelmattig) [inline services/src/contexts/migrations/migration_0003_gbif_config.rs:105]: either remove it or write some SQL that checks the provider survived the migration

File: services/src/contexts/migrations/migration_0004_dataset_listing_provider_prio.rs [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+        tx.batch_execute(
+            r#"
+            ALTER TABLE layer_providers
+            ADD COLUMN priority smallint NOT NULL DEFAULT 0;
+
+            ALTER TYPE "DatasetLayerListingProviderDefinition"
+            ADD ATTRIBUTE priority smallint;
+
+            ALTER TYPE "ArunaDataProviderDefinition"
+            ADD ATTRIBUTE description text;
```
Comment (michaelmattig) [inline services/src/contexts/migrations/migration_0004_dataset_listing_provider_prio.rs:32]: wouldn't it make sense to store data that is common to all data provider, in the layer_providers table?

File: services/src/layers/listing.rs [hunk 3]
```diff
@@ -94,6 +94,16 @@ pub trait LayerCollectionProvider {
...
 
+    /// get the name of the provider
+    fn name(&self) -> &str;
+
+    /// get the description of the provider
+    fn description(&self) -> &str;
+
+    /// get the priority of the provider
+    // higher priority providers are preferred over lower priority providers
+    // fn priority(&self) -> i8;
```
Comment (michaelmattig) [inline services/src/layers/listing.rs:105]: remove?

File: services/src/contexts/migrations/migration_0004_dataset_listing_provider_prio.rs [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+            ALTER TYPE "ArunaDataProviderDefinition"
+            ADD ATTRIBUTE priority smallint;
```
Comment (michaelmattig) [inline services/src/contexts/migrations/migration_0004_dataset_listing_provider_prio.rs:34]: this is already stored in the layer_providers table. We could remove it from the corresponding Db types and not store it in the database.

File: services/src/api/handlers/layers.rs [hunk 5]
```diff
@@ -187,6 +187,10 @@ async fn get_layer_providers<C: ApplicationContext>(
         })
         .await?
     {
+        if provider_listing.priority == i16::MIN {
```
Comment (michaelmattig) [inline services/src/api/handlers/layers.rs:190]: unrelated to this line: The Data Catalogue will always be on top, right? Is that ok?

File: test_data/provider_defs/user_datasets.json [hunk 6]
```diff
@@ -1,8 +1,9 @@
 {
   "type": "DatasetLayerListingProviderDefinition",
   "id": "cbb21ee3-d15d-45c5-a175-66964adf4e85",
-  "name": "User Data Listing",
-  "description": "User specific datasets grouped by tags.",
+  "name": "Personal Data Catalog",
```
Comment (michaelmattig) [inline test_data/provider_defs/user_datasets.json:4]: why did you change it? Was there already some feedback?

Comment (jdroenner) [reply test_data/provider_defs/user_datasets.json:4]: yes there is some feedback in Slack.

File: services/src/api/handlers/layers.rs (hunk 5 above)
Comment (jdroenner) [reply services/src/api/handlers/layers.rs:190]: yes it is. I think it should be always on top since it is the main entry point.

File: services/src/contexts/migrations/migration_0004_dataset_listing_provider_prio.rs (hunk 2 above)
Comment (jdroenner) [reply services/src/contexts/migrations/migration_0004_dataset_listing_provider_prio.rs:32]: maybe. It will require more logic so we decided to keep it like this.

File: services/src/layers/listing.rs (hunk 3 above)
Comment (jdroenner) [reply services/src/layers/listing.rs]: https://github.com/geo-engine/geoengine/pull/920/commits/3943d688b4c8460eb0f5315f13e5c2f93e4b563b

File: services/src/layers/listing.rs [hunk 7]
```diff
@@ -94,6 +94,16 @@ pub trait LayerCollectionProvider {
     /// Retrieve the provider's capabilities
     fn capabilities(&self) -> ProviderCapabilities;
 
+    /// get the name of the provider
+    fn name(&self) -> &str;
```
[the remark this answers is not in the document]
Comment (jdroenner) [reply services/src/layers/listing.rs:98]: they are used in the listing method

File: services/src/contexts/migrations/migration_0003_gbif_config.rs (hunk 1 above)
Comment (jdroenner) [reply services/src/contexts/migrations/migration_0003_gbif_config.rs]: [c3fe060](https://github.com/geo-engine/geoengine/pull/920/commits/c3fe060c999ac8b2bb6d30dbd1496247b74dff0e)

File: services/src/layers/postgres_layer_db.rs [hunk 8]
```diff
@@ -917,10 +940,10 @@ where
             SELECT 
                 id, 
                 name,
-                type_name
+                priority
             FROM 
                 layer_providers
-            ORDER BY name ASC
+            ORDER BY priority DESC, name ASC
```
Comment (michaelmattig) [inline services/src/layers/postgres_layer_db.rs:946]: I think here and in pro you should also proably do "WHERE priority > -1000", otherwise the user might get fewer than expected results, if they are only filtered out in the handler afterwards.

Comment (jdroenner) [reply services/src/layers/postgres_layer_db.rs:948]: https://github.com/geo-engine/geoengine/pull/920/commits/6cbdadc04008488233e0737326dab2f535e6db30

**Review by michaelmattig — APPROVED** [review]
